packages/react/src/components/Provider/Provider.tsx

```diff
@@ -37,192 +37,176 @@ export interface ProviderProps extends ChildrenComponentProps {
+      )
+    }
+  })
+}
+
+/**
+ * The Provider passes the CSS-in-JS renderer, theme styles and other settings to Fluent UI components.
+ */
+const Provider: React.FC<WithAsProp<ProviderProps>> & {
+  Box: typeof ProviderBox
+  Consumer: typeof ProviderConsumer
+  handledProps: (keyof ProviderProps)[]
+} = props => {
+  const { as, children, overwrite, variables, telemetryRef } = props
+  const unhandledProps = getUnhandledProps(Provider.handledProps, props)
+
+  const telemetry = React.useMemo<Telemetry | undefined>(() => {
+    if (!telemetryRef) {
+      return undefined
     }
-  }
 
-  componentWillUnmount() {
-    if (this.props.target) {
-      tryCleanupWhatInput(this.props.target)
+    if (!telemetryRef.current) {
+      telemetryRef.current = new Telemetry()
     }
+
+    return telemetryRef.current
+  }, [telemetryRef])
+  const inputContext: ProviderContextInput = {
+    theme: props.theme,
+    rtl: props.rtl,
+    disableAnimations: props.disableAnimations,
+    renderer: props.renderer,
+    target: props.target,
+    telemetry,
   }
 
-  render() {
-    const {
-      as,
-      children,
-      disableAnimations,
-      overwrite,
-      renderer,
-      rtl,
-      target,
-      theme,
-      variables,
-      telemetryRef,
-      ...unhandledProps
-    } = this.props
-
-    if (telemetryRef) {
-      if (!this.telemetry) {
-        this.telemetry = new Telemetry()
-      }
+  const consumedContext: ProviderContextPrepared = React.useContext(ThemeContext)
+  const incomingContext: ProviderContextPrepared = overwrite ? ({} as any) : consumedContext
 
-      telemetryRef['current'] = this.telemetry
-    } else if (this.telemetry) {
-      delete this.telemetry
-    }
+  // rehydration disabled to avoid leaking styles between renderers
+  // https://github.com/rofrischmann/fela/blob/master/docs/api/fela-dom/rehydrate.md
+  const outgoingContext = mergeContexts(incomingContext, inputContext)
 
-    const inputContext: ProviderContextInput = {
-      theme,
-      rtl,
-      disableAnimations,
-      renderer,
-      target,
-      telemetry: this.telemetry,
-    }
+  const rtlProps: { dir?: 'rtl' | 'ltr' } = {}
+  // only add dir attribute for top level provider or when direction changes from parent to child
+  if (
+    !consumedContext ||
+    (consumedContext.rtl !== outgoingContext.rtl && _.isBoolean(outgoingContext.rtl))
+  ) {
+    rtlProps.dir = outgoingContext.rtl ? 'rtl' : 'ltr'
+  }
 
-    const incomingContext: ProviderContextPrepared = overwrite ? {} : this.context
-    // rehydration disabled to avoid leaking styles between renderers
-    // https://github.com/rofrischmann/fela/blob/master/docs/api/fela-dom/rehydrate.md
-    this.outgoingContext = mergeContexts(incomingContext, inputContext)
+  React.useLayoutEffect(() => {
+    renderFontFaces(outgoingContext.renderer, props.theme)
+    renderStaticStyles(outgoingContext.renderer, props.theme, outgoingContext.theme.siteVariables)
```

Comment:
This is an important change I think. Previously we rendered `fontFaces` in `componentDidMount()`, but `staticStyles` during render. 

Is there a specific requirement to do it as it was?